wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/SyntaxRules.kt

```diff
@@ -21,6 +21,7 @@ import com.squareup.wire.schema.ProtoFile.Syntax.PROTO_3
 
 /** A set of rules which defines schema requirements for a specific [Syntax]. */
 interface SyntaxRules {
+  fun allowUserDefinedDefaultValue(): Boolean
```

Comment:
I don't know for sure how much logic `SyntaxRules` can have. i.e. Is it good to pass `userDefinedDefaultValue: String?` to the func to compute the result?

In this case, do we want to check that
a) can I have a user-defined default value? e.g. `allowUserDefinedDefaultValue()` + is this default value set by the user?
or
b) can I use this default value? e.g. `allowUserDefinedDefaultValue(defaultValue: String?)` or `canUseUserDefinedDefaultValue(defaultValue: String?)`

Reply:
I like a better in this case.